File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/main/CloudAccountsView.java
package app.editors.manager.mvp.views.main;

import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;

import java.util.List;

import app.editors.manager.mvp.models.account.AccountsSqlData;
import app.editors.manager.mvp.views.base.BaseView;

@StateStrategyType(OneExecutionStateStrategy.class)
public interface CloudAccountsView extends BaseView {

    void onAccountLogin();

    void onWebDavLogin(AccountsSqlData account);

    void onShowClouds();

    void onShowBottomDialog(AccountsSqlData account);

    void onSetAccounts(List<AccountsSqlData> accounts);

    void onShowWaitingDialog();

    void removeItem(int position);

    void onUpdateItem(AccountsSqlData account);

    void onSuccessLogin();

    void onSignIn(String portal, String login);

    void onEmptyList();

    void onSelectionMode();

    void onDefaultState();

    void onSelectedItem(int position);

    void onActionBarTitle(String title);

    void onNotifyItems();
}

File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/main/CloudAccountsView.kt
package app.editors.manager.mvp.views.main

import app.documents.core.storage.account.CloudAccount
import app.editors.manager.mvp.presenters.main.CloudAccountState
import app.editors.manager.mvp.views.base.BaseView
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(OneExecutionStateStrategy::class)
interface CloudAccountsView : BaseView {
    fun onAccountLogin()
    fun onWebDavLogin(account: CloudAccount)
    fun onShowClouds()
    fun onShowBottomDialog(account: CloudAccount?)
    fun onShowWaitingDialog()
    fun removeItem(position: Int)
    fun onUpdateItem(account: CloudAccount, position: Int)
    fun onSuccessLogin()
    fun onSignIn(portal: String?, login: String)
    fun onEmptyList()
    fun onSelectionMode()
    fun onDefaultState()
    fun onSelectedItem(position: Int)
    fun onActionBarTitle(title: String)
    fun onNotifyItems()
    fun onRender(state: CloudAccountState)
}